packages/fluentui/react-northstar/src/components/Datepicker/Datepicker.tsx

```diff
@@ -100,6 +100,14 @@ export interface DatepickerProps extends UIComponentProps, Partial<ICalendarStri
   /** Controls the calendar's open state. */
   calendarOpenState?: boolean;
 
+  /**
+   * Called on change of the open state.
+   *
+   * @param event - React's original SyntheticEvent.
+   * @param data - All props and proposed value.
+   */
+  onCalendarOpenStateChange?: ComponentEventHandler<DatepickerProps & { value: Date }>;
```

Comment:
```suggestion
  onCalendarOpenStateChange?: ComponentEventHandler<DatepickerProps>;
```

you're not actually setting the date in here. From I see it's just spreading props and including the new value of the calendarOpenState